app/models/miq_storage_metric.rb

```diff
@@ -84,24 +84,19 @@ def daily_rollup_by_type(rollup_time, time_profile, metric_list)
   end
 
   def derived_metrics_in_range(start_time, end_time)
-    cond = ["statistic_time >= ? AND statistic_time < ?", start_time, end_time]
-    miq_derived_metrics.where(cond).to_a
+    miq_derived_metrics.where(:statistic_time => start_time..end_time).to_a
```

Comment:
Sorry for killing the fun, but this refactoring slightly changes the behavior of the method. `BETWEEN` includes both endpoints (at least in PostgreSQL and SQLite), but in the original code only start_time was included and end_time was not. I don't know whether this is a problem or not. Just something to be aware of.